torch/csrc/jit/codegen/cuda/compute_at_map.cpp

```diff
@@ -292,11 +292,21 @@ void ComputeAtMap::build(Fusion* fusion, GpuLower* gpu_lower) {
         for (auto entry : c2p_map) {
           auto c_id = entry.first;
           auto p_id = entry.second;
-          // If outside CA point and we're creating parallel map, do not map the
-          // axis
-          if (mapping_mode_ == MappingMode::PARALLEL &&
-              right_of_ca_point.find(p_id) != right_of_ca_point.end()) {
-            continue;
+          // If we're creating parallel map, only map the leaf
+          // axes. Also, the producer axis must be left of the CA
+          // point.
+          if (mapping_mode_ == MappingMode::PARALLEL) {
```

Comment:
Sorry, maybe wasn't clear. I think you should pull this out of the for loop. We can just iterate over `c_tv->domain()` and check the entry is in the map not the other way around. As `c_tv->domain()` is always smaller than `c2p_map`. Fairly minor optimization so approving anyways at this point.

Reply:
Thanks. Updated.